packages/nova-react/src/eventing/nova-eventing-provider.tsx

```diff
@@ -133,20 +192,37 @@ const generateEventing =
     mapperRef: React.MutableRefObject<
       (reactEventWrapper: ReactEventWrapper) => EventWrapper
     >,
+    interceptorRef?: React.MutableRefObject<
+    (event: EventWrapper) => Promise<EventWrapper | undefined>
+  >,
   ) =>
   (): NovaReactEventing => ({
-    bubble: (eventWrapper: ReactEventWrapper) => {
+    bubble: async (eventWrapper: ReactEventWrapper) => {
       const mappedEvent: EventWrapper = mapperRef.current(eventWrapper);
-      return eventingRef.current.bubble(mappedEvent);
+      if (!interceptorRef) {
+        return eventingRef.current.bubble(mappedEvent);
+      }
+
+      let eventToBubble: EventWrapper | undefined = mappedEvent;
+      eventToBubble = await interceptorRef?.current(mappedEvent);
```

Comment:
When can `eventToBubble` be null/undefined here?

Reply:
Should we return undefined from interceptor handler to make sure the processed event is not bubbled up? And otherwise just return event it got, right?